pandas/core/frame.py

```diff
@@ -5145,36 +5145,151 @@ def pivot_table(self, values=None, index=None, columns=None,
+        index with one or more new inner-most levels compared to the current
+        dataframe. The new inner-most levels are created by pivoting the
+        columns of the current dataframe:
+
+          - if the columns have a single level, the output is a Series;
+          - if the columns have multiple levels, the new index
+            level(s) is (are) taken from the prescribed level(s) and
+            the output is a DataFrame.
+
+        The new index levels are sorted.
 
         Parameters
         ----------
-        level : int, string, or list of these, default last level
-            Level(s) to stack, can pass level name
-        dropna : boolean, default True
-            Whether to drop rows in the resulting Frame/Series with no valid
-            values
+        level : int, str, list, default -1
+            Level(s) to stack from the column axis onto the index
+            axis, defined as one index or label, or a list of indices
+            or labels.
+        dropna : bool, default True
+            Whether to drop rows in the resulting Frame/Series with
+            missing values. Stacking a column level onto the index
+            axis can create combinations of index and column values
+            that are missing from the original dataframe. See Examples
+            section.
+
+        Notes
+        -----
+        The function is named by analogy with a stack of books
+        (levels) being re-organised from a horizontal position (column
+        levels) to a vertical position (index levels).
 
         Examples
-        ----------
-        >>> s
+        --------
+        >>> df_single_level_cols = pd.DataFrame([[0, 1], [2, 3]],
+        ...                                     index=['one', 'two'],
+        ...                                     columns=['a', 'b'])
+        >>> multicol1 = pd.MultiIndex.from_tuples([('X', 'a'), ('X', 'b')])
+        >>> df_multi_level_cols1 = pd.DataFrame([[0, 1], [2, 3]],
+        ...                                    index=['one', 'two'],
+        ...                                    columns=multicol1)
+        >>> multicol2 = pd.MultiIndex.from_tuples([('X', 'a'), ('Y', 'b')])
+        >>> df_multi_level_cols2 = pd.DataFrame([[0.0, 1.0], [2.0, 3.0]],
+        ...                                     index=['one', 'two'],
+        ...                                     columns=multicol2)
+        >>> df_multi_level_cols3 = pd.DataFrame([[None, 1.0], [2.0, 3.0]],
+        ...                                     index=['one', 'two'],
+        ...                                     columns=multicol2)
+
+        Stacking a dataframe with a single level column axis returns a Series:
+
+        >>> df_single_level_cols
              a   b
-        one  1.  2.
-        two  3.  4.
+        one  0   1
+        two  2   3
+        >>> df_single_level_cols.stack()
+        one  a    0
+             b    1
+        two  a    2
+             b    3
+        dtype: int64
```

Comment:
It's just a personal opinion, but I think defining all the data first with descriptive names make it a bit more complex to understand.

We could have separate sections for each case, with a title in bold (surrounding the text with double stars, followed by the data creation, using simply `df` in all the cases.

Also, in this case I think it would make the example easier to understand using more real-world examples. As `a`, `b`... don't have a meaning, it'd a bit harder to understand what's going on.

A minor thing, when creating the data, I think it makes more sense that each row is defined as a tuple, than as a list.

For example:

```
**Single level**

>>> df = pd.DataFrame([(8, 12), (22, 35)],
...                   index=['cat', 'dog'],
...                   columns=['weight', 'max_speed'])
>>> df

>>> df.stack()
```

Reply:
:+1: fixed in 15902ed